### PR TITLE
chore: Update `top` with `events_in_total` and `events_out_total` metrics

### LINKED ISSUE #7257: Update `top` with `events_in_total` and `events_out_total` metrics
`vector top` should replace it's usage of `events_processed_total` metric with one or both of `events_in_total` and `events_out_total` metrics which are replacing `events_processed_total`. 

ref #6367

### PR DESCRIPTION
**Pull request by ktff**
Closes #7257 

Screenshot from 2021-04-30 16-55-43

### CODE REVIEW

**Comment by ktff** [comment]
I wanted to retain one word description per column that we had, but yes, it would be clearer. I'll change it.

File: src/top/metrics.rs [hunk 1]
```diff
@@ -49,47 +51,84 @@ async fn component_removed(client: Arc<SubscriptionClient>, tx: state::EventTx)
     }
 }
 
-async fn processed_events_totals(
-    client: Arc<SubscriptionClient>,
-    tx: state::EventTx,
-    interval: i64,
-) {
-    let res = client.component_processed_events_totals_subscription(interval);
+async fn events_in_total(client: Arc<SubscriptionClient>, tx: state::EventTx, interval: i64) {
```
Comment (leebenson) [inline src/top/metrics.rs:54]:
```suggestion
async fn events_in_totals(client: Arc<SubscriptionClient>, tx: state::EventTx, interval: i64) {
```

Nit: Similar functions use the plural form.

File: src/top/metrics.rs [hunk 2]
```diff
@@ -154,12 +193,14 @@ pub fn subscribe(client: SubscriptionClient, tx: state::EventTx, interval: i64)
 
     tokio::spawn(component_added(Arc::clone(&client), tx.clone()));
     tokio::spawn(component_removed(Arc::clone(&client), tx.clone()));
-    tokio::spawn(processed_events_totals(
+    tokio::spawn(events_in_total(Arc::clone(&client), tx.clone(), interval));
```
Comment (leebenson) [inline src/top/metrics.rs:196]:
```suggestion
    tokio::spawn(events_in_totals(Arc::clone(&client), tx.clone(), interval));
```

To match the previous suggestion.

File: src/top/metrics.rs [hunk 3]
```diff
@@ -49,47 +51,84 @@ async fn component_removed(client: Arc<SubscriptionClient>, tx: state::EventTx)
...
+                .send(state::EventType::EventsInThroughputs(
+                    interval,
+                    c.into_iter().map(|c| (c.name, c.throughput)).collect(),
+                ))
+                .await;
+        }
+    }
+}
+
+async fn events_out_total(client: Arc<SubscriptionClient>, tx: state::EventTx, interval: i64) {
```
Comment (leebenson) [inline src/top/metrics.rs:95]:
```suggestion
async fn events_out_totals(client: Arc<SubscriptionClient>, tx: state::EventTx, interval: i64) {
```

Nit: Plural form.

File: src/top/metrics.rs [hunk 4]
```diff
@@ -154,12 +193,14 @@ pub fn subscribe(client: SubscriptionClient, tx: state::EventTx, interval: i64)
...
     tokio::spawn(component_removed(Arc::clone(&client), tx.clone()));
-    tokio::spawn(processed_events_totals(
+    tokio::spawn(events_in_total(Arc::clone(&client), tx.clone(), interval));
+    tokio::spawn(events_in_throughputs(
         Arc::clone(&client),
         tx.clone(),
         interval,
     ));
-    tokio::spawn(processed_events_throughputs(
+    tokio::spawn(events_out_total(Arc::clone(&client), tx.clone(), interval));
```
Comment (leebenson) [inline src/top/metrics.rs:202]:
```suggestion
    tokio::spawn(events_out_totals(Arc::clone(&client), tx.clone(), interval));
```

**Comment by ktff** [comment]
This is how it looks on 81 char width. It's crowded but the important parts are readable.

Screenshot from 2021-04-30 16-51-28

**Review by leebenson — CHANGES_REQUESTED** [review]
Looks great 👍🏻 Just some very minor nits re: pluralising function names to match.

**Review by leebenson — APPROVED** [review]

**Comment by jszwedko** [comment]
This would be another case where a two-level heading would be useful to do something like:

```
   Events
  In    Out
```

But I can't remember if the term library we are using supports that.

**Review by jszwedko — APPROVED** [review]
Very cool, thanks @ktff ! I'm happy to see `top` getting more useful.

**Comment by jszwedko** [comment]
We could maybe call `Bytes` `Bytes Out` just to make that clearer.

**Comment by ktff** [comment]
Just checked and, there doesn't seem to be a way to have two level header in `tui`.

> We could maybe call Bytes Bytes Out just to make that clearer.

Bytes is intentionally ambiguous since for sources that is In, but for sinks that is Out.

**Pull request merged**